### PR TITLE
Gsf plotting

### PR DESCRIPTION
**Pull request by lispandfound**
Adds a utility to plot a GSF file. Useful for debugging output of utilities like `fault_seg2gsf` or the new python gsf generator.

### CODE REVIEW

File: sources/gsf_plot.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        "hot",
+        (0, points["depth"].max(), 1),
+        ("white", "black"),
+        plot_contours=False,
+    )
+
+    return fig
+
+
+@app.command(help="Plot a GSF file using GMT.")
```
Comment (sungeunbae) [inline sources/gsf_plot.py:150]: Interesting use of typer, typing.Annotated.

**Review by sungeunbae — COMMENTED** [review]
The use of typer and typing makes this script quite uniquely styled. As this is a stand-alone script, I don't see a problem.

**Review by sungeunbae — APPROVED** [review]

File: sources/gsf_plot.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+def plot_gsf_points(points: pd.DataFrame, grid_resolution: int) -> pygmt.Figure:
+    """Plot a set of GSF points (lat, lon, depth) in a map using PyGMT.
+
+    Parameters
+    ----------
+    points : pd.DataFrame
+        The GSF points to plot. DataFrame must at least contain the lat, lon and depth columns.
+    grid_resolution : int
+        The resolution of the grid. If this is smaller than the resolution
```
Comment (joelridden) [inline sources/gsf_plot.py:115]: Not a huge deal but maybe mention this is in meters like the script below description just to understand the expected values to use with this function.

**Review by joelridden — COMMENTED** [review]
Maybe a image in the description of the PR of an example output plot would be good for context

**Comment by lispandfound** [comment]
By popular demand, an example of what a GSF file _might_ look like. It even has a buggy title! (will fix this).

plot

**Review by joelridden — APPROVED** [review]

**Review by AndrewRidden-Harper — APPROVED** [review]

**Comment by lispandfound** [comment]
This PR needs some work. The bounds of the geometry are a little harder to discern than I thought they would be (hence the weird corners in the example). The `alphashape` package is rather buggy and doesn't do what I need it to do (pull out the corners of the geometry). I have an algorithm in mind that should work, however.

**Comment by sungeunbae** [comment]
> This PR needs some work. The bounds of the geometry are a little harder to discern than I thought they would be (hence the weird corners in the example). The `alphashape` package is rather buggy and doesn't do what I need it to do (pull out the corners of the geometry). I have an algorithm in mind that should work, however.

Interesting comment about alphashape - not sure in what context you are considering alphashape, but when you think about it, see if you could improve (replace with a better algorithm) this bit too. https://github.com/ucgmsim/qcore/blob/f57aae21c1489315369c9791a851c991ffb06c4c/qcore/srf.py#L827

File: sources/gsf_plot.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+import pandas as pd
+import pygmt
+import typer
+from pygmt_helper import plotting
+
+app = typer.Typer()
+
+
+def parse_gsf(gsf_filepath: str) -> pd.DataFrame:
```
Comment (claudio525) [inline sources/gsf_plot.py:31]: Wasn't the plan to merge this https://github.com/ucgmsim/qcore/pull/303 and then use the parsing function from qcore?

Comment (lispandfound) [reply sources/gsf_plot.py]: yeah that's an old commit it is using qcore now.

**Comment by lispandfound** [comment]
The algorithm has now been updated to fix the corner issue and make the plot output align closely with the plot srf output. You can see the new output here:

plot

@sungeunbae regarding the SRF perimeter code, I do think alphashape is the wrong approach but I need to properly understand roughness before I can pass too much judgement.

**Review by claudio525 — APPROVED** [review]

**Review by sungeunbae — APPROVED** [review]

**Pull request merged**